src/services/sourcemaps.ts

```diff
@@ -0,0 +1,330 @@
+            }
+            return file;
+        }
+
+        function getPositionOfLineAndCharacterUsingName(fileName: string, directory: string, line: number, character: number) {
+            const file = getSourceFileLike(fileName, directory);
+            if (!file) {
+                return -1;
+            }
+            return getPositionOfLineAndCharacter(file, line, character);
+        }
+
+        function getDecodedMappings() {
+            return decodedMappings || (decodedMappings = calculateDecodedMappings());
+        }
+
+        function getReverseSortedMappings() {
+            return reverseSortedMappings || (reverseSortedMappings = getDecodedMappings().slice().sort(compareProcessedSpanSourcePositions));
+        }
+
+        function getForwardSortedMappings() {
+            return forwardSortedMappings || (forwardSortedMappings = getDecodedMappings().slice().sort(compareProcessedSpanEmittedPositions));
+        }
+
+        function calculateDecodedMappings(): ProcessedSourceMapSpan[] {
+            const state: DecoderState<ProcessedSourceMapSpan> = {
+                encodedText: map.mappings,
+                currentNameIndex: undefined,
+                sourceMapNamesLength: map.names ? map.names.length : undefined,
+                currentEmittedColumn: 1,
+                currentEmittedLine: 1,
+                currentSourceColumn: 1,
+                currentSourceLine: 1,
+                currentSourceIndex: 0,
+                spans: [],
+                decodingIndex: 0,
+                processSpan,
+            };
+            while (!hasCompletedDecoding(state)) {
+                decodeSingleSpan(state);
+                if (state.error) {
+                    host.log(`Encountered error while decoding sourcemap found at ${mapPath}: ${state.error}`);
+                    return [];
+                }
+            }
+            return state.spans;
+        }
+
+        function compareProcessedSpanSourcePositions(a: ProcessedSourceMapSpan, b: ProcessedSourceMapSpan) {
+            return comparePaths(a.sourcePath, b.sourcePath, sourceRoot) ||
+                compareValues(a.sourcePosition, b.sourcePosition);
+        }
+
+        function compareProcessedSpanEmittedPositions(a: ProcessedSourceMapSpan, b: ProcessedSourceMapSpan) {
+            return compareValues(a.emittedPosition, b.emittedPosition);
+        }
+
+        function processSpan(span: RawSourceMapSpan): ProcessedSourceMapSpan {
+            const sourcePath = map.sources[span.sourceIndex];
+            return {
+                emittedPosition: getPositionOfLineAndCharacterUsingName(map.file, currentDirectory, span.emittedLine - 1, span.emittedColumn - 1),
+                sourcePosition: getPositionOfLineAndCharacterUsingName(sourcePath, sourceRoot, span.sourceLine - 1, span.sourceColumn - 1),
+                sourcePath,
+                name: span.nameIndex ? map.names[span.nameIndex] : undefined

```

Comment:
We don't actually use names anywhere in the results, so this is unnecessary work.